integration-libs/cdc/core/profile/facade/facade-providers.ts

```diff
@@ -0,0 +1,31 @@
+import { Provider } from '@angular/core';
+import { UserEmailService, UserPasswordService, UserProfileService } from '@spartacus/user/profile/core';
+import {
+  UserEmailFacade,
+  UserPasswordFacade,
+  UserProfileFacade,
+  UserRegisterFacade
+} from '@spartacus/user/profile/root';
+import { CDCUserRegisterService } from 'integration-libs/cdc/core/profile/facade/cdc-user-register.service';
+export const facadeProviders: Provider[] = [
+  UserEmailService,
+  UserPasswordService,
+  UserProfileService,
+  CDCUserRegisterService,
+  {
+    provide: UserEmailFacade,
+    useExisting: UserEmailService,
+  },
+  {
+    provide: UserPasswordFacade,
+    useExisting: UserPasswordService,
+  },
+  {
+    provide: UserProfileFacade,
+    useExisting: UserProfileService,
+  },
```

Comment:
Do we need to provide originals? (and UserProfileConnector original too )? Let's double check.

Reply:
Hi @Platonn,

While removing the `UserProfileService` from the file, I get the error in runtime -

The register page does not load.

```
core.mjs:6485 ERROR Error: NullInjectorError: No provider for UserProfileService!
    at NullInjector.get (core.mjs:11120:1) [angular]
    at R3Injector.get (core.mjs:11287:1) [angular]
    at R3Injector.get (core.mjs:11287:1) [angular]
    at R3Injector.get (core.mjs:11287:1) [angular]
    at R3Injector.get (core.mjs:11287:1) [angular]
    at injectInjectorOnly (core.mjs:4765:1) [angular]
    at Module.ɵɵinject (core.mjs:4769:1) [angular]
    at Object.CDCUserRegisterService_Factory [as factory] (cdc-user-register.service.ts:22:36) [angular]
    at R3Injector.hydrate (core.mjs:11457:1) [angular]
    at R3Injector.get (core.mjs:11276:1) [angular]
    at injectInjectorOnly (core.mjs:4765:1) [angular]
    at ɵɵinject (core.mjs:4769:1) [angular]
    at Object.factory (core.mjs:11544:1) [angular]
    at R3Injector.hydrate (core.mjs:11457:1) [angular]
defaultErrorLogger @ core.mjs:6485
handleError @ core.mjs:6532
next @ core.mjs:26046
__tryOrUnsub @ Subscriber.js:183
next @ Subscriber.js:122
_next @ Subscriber.js:72
next @ Subscriber.js:49
next @ Subject.js:39
emit @ core.mjs:22423
(anonymous) @ core.mjs:25514
invoke @ zone.js:372
run @ zone.js:134
runOutsideAngular @ core.mjs:25387
onHandleError @ core.mjs:25514
handleError @ zone.js:376
runTask @ zone.js:181
invokeTask @ zone.js:487
ZoneTask.invoke @ zone.js:476
data.args.<computed> @ zone.js:2541
setTimeout (async)
scheduleTask @ zone.js:2543
scheduleTask @ zone.js:393
onScheduleTask @ zone.js:283
scheduleTask @ zone.js:386
scheduleTask @ zone.js:221
scheduleMacroTask @ zone.js:244
scheduleMacroTaskWithCurrentZone @ zone.js:679
(anonymous) @ zone.js:2585
proto.<computed> @ zone.js:975
hostReportError @ hostReportError.js:2
error @ Observer.js:11
__tryOrUnsub @ Subscriber.js:183
error @ Subscriber.js:135
_error @ Subscriber.js:75
error @ Subscriber.js:55
_error @ Subscriber.js:75
error @ Subscriber.js:55
_error @ tap.js:56
error @ Subscriber.js:55
_trySubscribe @ Observable.js:50
subscribe @ Observable.js:28
call @ tap.js:16
subscribe @ Observable.js:23
call @ finalize.js:11
subscribe @ Observable.js:23
launchComponent @ component-wrapper.directive.ts:110
(anonymous) @ component-wrapper.directive.ts:73
__tryOrUnsub @ Subscriber.js:183
next @ Subscriber.js:122
_next @ Subscriber.js:72
next @ Subscriber.js:49
(anonymous) @ subscribeToArray.js:3
_trySubscribe @ Observable.js:42
subscribe @ Observable.js:28
(anonymous) @ defer.js:15
_trySubscribe @ Observable.js:42
subscribe @ Observable.js:28
ngOnInit @ component-wrapper.directive.ts:67
callHook @ core.mjs:2542
callHooks @ core.mjs:2511
executeInitAndCheckHooks @ core.mjs:2462
refreshView @ core.mjs:9499
refreshEmbeddedViews @ core.mjs:10609
refreshView @ core.mjs:9508
refreshEmbeddedViews @ core.mjs:10609
refreshView @ core.mjs:9508
refreshEmbeddedViews @ core.mjs:10609
refreshView @ core.mjs:9508
refreshComponent @ core.mjs:10655
refreshChildComponents @ core.mjs:9280
refreshView @ core.mjs:9534
refreshEmbeddedViews @ core.mjs:10609
refreshView @ core.mjs:9508
refreshEmbeddedViews @ core.mjs:10609
refreshView @ core.mjs:9508
refreshComponent @ core.mjs:10655
refreshChildComponents @ core.mjs:9280
refreshView @ core.mjs:9534
refreshEmbeddedViews @ core.mjs:10609
refreshView @ core.mjs:9508
refreshEmbeddedViews @ core.mjs:10609
refreshView @ core.mjs:9508
refreshComponent @ core.mjs:10655
refreshChildComponents @ core.mjs:9280
refreshView @ core.mjs:9534
renderComponentOrTemplate @ core.mjs:9598
tickRootContext @ core.mjs:10829
detectChangesInRootView @ core.mjs:10854
detectChanges @ core.mjs:21439
tick @ core.mjs:26449
(anonymous) @ core.mjs:26304
invoke @ zone.js:372
onInvoke @ core.mjs:25488
invoke @ zone.js:371
run @ zone.js:134
run @ core.mjs:25342
next @ core.mjs:26303
__tryOrUnsub @ Subscriber.js:183
next @ Subscriber.js:122
_next @ Subscriber.js:72
next @ Subscriber.js:49
next @ Subject.js:39
emit @ core.mjs:22423
checkStable @ core.mjs:25410
onHasTask @ core.mjs:25505
hasTask @ zone.js:426
_updateTaskCount @ zone.js:447
_updateTaskCount @ zone.js:274
runTask @ zone.js:195
drainMicroTaskQueue @ zone.js:582
Promise.then (async)
scheduleMicroTask @ zone.js:565
scheduleTask @ zone.js:396
onScheduleTask @ zone.js:283
scheduleTask @ zone.js:386
scheduleTask @ zone.js:221
scheduleMicroTask @ zone.js:241
scheduleResolveOrReject @ zone.js:1266
resolvePromise @ zone.js:1204
(anonymous) @ zone.js:1120
webpackJsonpCallback @ jsonp chunk loading:77
(anonymous) @ default-feature-libs_user_profile_components_public_api_ts-feature-libs_user_profile_occ_publ-4d6199.js:2
Show 79 more frames
```